src/dataframe/dataframe.jl

```diff
@@ -1410,3 +1410,67 @@ end
 function permutecols!(df::DataFrame, p::AbstractVector{Symbol})
     permutecols!(df, index(df)[p])
 end
+
+##############################################################################
+##
+## Expand & Complete dataframes
+##
+##############################################################################
+
+function _expandhelper(col, iterprod, idx)
+    for v in iterprod
+        push!(col, v[idx])
+    end
+end
+
+function expand(df::AbstractDataFrame, indexcols; error::Bool=true, complete::Bool=false, fill=missing, replaceallmissing::Bool=false)
+    colind = index(df)[indexcols]
+    dummydf = similar(select(df, colind, copycols=false), 0)
+
+    if row_group_slots(ntuple(i -> df[!,colind][!, i], ncol(df[!,colind])), Val(false))[1] != nrow(df[!,colind])
+        if error == true
```

Comment:
I think it should be:
```
if error && !complete
```
because if `complete == true` then duplicate rows are not a problem (they will not be dropped as `join` will reintroduce them).